File: Excepciones/Ejercicio 2/Calculadora.py
class NumeroInvalidoException(Exception):
    pass

class Calculadora:

    @staticmethod
    def sumar(a, b):
        return a + b

    @staticmethod
    def restar(a, b):
        return a - b

    @staticmethod
    def multiplicar(a, b):
        return a * b

    @staticmethod
    def dividir(a, b):
        if b == 0:
            raise ZeroDivisionError("No se puede dividir entre cero.")
        return a / b

    @staticmethod
    def convertir_a_entero(valor):
        try:
            return int(valor)
        except ValueError:
            raise NumeroInvalidoException(f"'{valor}' no es un número válido.")
def main():
    try:
        num1 = Calculadora.convertir_a_entero("50")
        print("Conversión correcta:", num1)
    except NumeroInvalidoException as e:
        print("Error:", e)

    try:
        num2 = Calculadora.convertir_a_entero("abc")
        print("Conversión correcta:", num2)
    except NumeroInvalidoException as e:
        print("Error:", e)

    try:
        print("Suma:", Calculadora.sumar(10, 5))
        print("Resta:", Calculadora.restar(10, 5))
        print("Multiplicación:", Calculadora.multiplicar(10, 5))
        print("División:", Calculadora.dividir(10, 0))  
    except ZeroDivisionError as e:
        print("Error en división:", e)

    try:
        print("División:", Calculadora.dividir(50, 2))
    except ZeroDivisionError as e:
        print("Error en división:", e)

main()

File: Excepciones/Ejercicio 2/Calculadora.java
class Calculadora {

    public static int sumar(int a, int b) {
        return a + b;
    }

    public static int restar(int a, int b) {
        return a - b;
    }

    public static int multiplicar(int a, int b) {
        return a * b;
    }

    public static double dividir(int a, int b) {
        if (b == 0) {
            throw new ArithmeticException("No se puede dividir entre cero.");
        }
        return (double) a / b;
    }

    public static int convertirAEntero(String valor) throws NumeroInvalidoException {
        try {
            return Integer.parseInt(valor);
        } catch (NumberFormatException e) {
            throw new NumeroInvalidoException("El valor '" + valor + "' no es un número válido.");
        }
    }
}
